Microsoft.Toolkit.Uwp.UI.Media/Geometry/Utils.cs

```diff
@@ -0,0 +1,905 @@
+        /// </returns>
+        public static bool IsZero(this float value)
+        {
+            return Math.Abs(value) < 10.0f * FloatMin;
+        }
+
+        /// <summary>
+        /// Compares two points for fuzzy equality.  This function
+        /// helps compensate for the fact that double values can
+        /// acquire error when operated upon
+        /// </summary>
+        /// <param name='point1'>The first point to compare</param>
+        /// <param name='point2'>The second point to compare</param>
+        /// <returns>Whether or not the two points are equal</returns>
+        public static bool IsCloseTo(this Point point1, Point point2)
+        {
+            return point1.X.IsCloseTo(point2.X) && point1.Y.IsCloseTo(point2.Y);
+        }
+
+        /// <summary>
+        /// Compares two Size instances for fuzzy equality.  This function
+        /// helps compensate for the fact that double values can
+        /// acquire error when operated upon
+        /// </summary>
+        /// <param name='size1'>The first size to compare</param>
+        /// <param name='size2'>The second size to compare</param>
+        /// <returns>Whether or not the two Size instances are equal</returns>
+        public static bool IsCloseTo(this Size size1, Size size2)
+        {
+            return size1.Width.IsCloseTo(size2.Width) && size1.Height.IsCloseTo(size2.Height);
+        }
+
+        /// <summary>
+        /// Compares two rectangles for fuzzy equality.  This function
+        /// helps compensate for the fact that double values can
+        /// acquire error when operated upon
+        /// </summary>
+        /// <param name='rect1'>The first rectangle to compare</param>
+        /// <param name='rect2'>The second rectangle to compare</param>
+        /// <returns>Whether or not the two rectangles are equal</returns>
+        public static bool IsCloseTo(this Rect rect1, Rect rect2)
+        {
+            // If they're both empty, don't bother with the double logic.
+            if (rect1.IsEmpty)
+            {
+                return rect2.IsEmpty;
+            }
+
+            // At this point, rect1 isn't empty, so the first thing we can test is rect2.IsEmpty, followed by property-wise compares.
+            return (!rect2.IsEmpty)
+                   && rect1.X.IsCloseTo(rect2.X) && rect1.Y.IsCloseTo(rect2.Y)
+                   && rect1.Height.IsCloseTo(rect2.Height) && rect1.Width.IsCloseTo(rect2.Width);
+        }
+
+        /// <summary>
+        /// Restricts the given single precision number to be within the
+        /// specified min and max values (inclusive).
+        /// </summary>
+        /// <param name="num">Given single precision number</param>
+        /// <param name="min">Minimum value</param>
+        /// <param name="max">Maximum value</param>
+        /// <returns>Clamped value</returns>
+        public static float Clamp(this float num, float min, float max)
```

Comment:
A lot of these helpers are available in .NET 5, wonder if we can mark these for updating later...

Do we want to be exposing these all? @Sergio0694? If so, we could probably move them up the Toolkit path?

Reply:
Removed the Clamp methods since they are already in .NET 5.